app/models/user.js

```diff
@@ -41,5 +41,18 @@ module.exports = (sequelize, DataTypes) => {
       throw databaseError('Unable to create new user.');
     });
 
+  User.findByEmail = email =>
+    User.findOne({ where: { email } })
+      .then(response => response && response.dataValues)
+      .catch(error => {
+        logger.error(error);
+        const { message, errorFn } = dbErrorCodes[error.name] || { message: 'Unable to find user.' };
+        if (errorFn) {
+          throw errorFn(`Unable to find user. ${message}`);
+        }
+        logger.error(message);
+        throw databaseError('Unable to find user.');
```

Comment:
I saw similar code before, any way of avoid repeat this in each model function ?